include/nlohmann/detail/input/binary_reader.hpp

```diff
@@ -2411,14 +2424,23 @@ class binary_reader
         // detect and encode bjdata ndarray as an object in JData annotated array format (https://github.com/NeuroJSON/jdata):
         // {"_ArrayType_" : "typeid", "_ArraySize_" : [n1, n2, ...], "_ArrayData_" : [v1, v2, ...]}
 
-        if (input_format == input_format_t::bjdata && size_and_type.first != string_t::npos && size_and_type.first >= (1ull << (sizeof(std::size_t) * 8 - 1)))
+        if (input_format == input_format_t::bjdata && size_and_type.first != string_t::npos && size_and_type.second >= (1 << 8))
```

Comment:
Use this to test if a bit is set:

```cpp
if (input_format == input_format_t::bjdata
    && size_and_type.first != string_t::npos
    && static_cast<bool>(size_and_type.second & (1 << 8)))
```

Alternatively, explicitly check for non-zero:
```cpp
(size_and_type.second & (1 << 8)) != 0
```